broadinstitute/consent
# [DUOS-1582][risk=no] Java 17 Updates

## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1582

Minor updates to build/run under java 17

See also: https://github.com/DataBiosphere/consent-ontology/pull/668

## Testing
1. Download and install java 17 in some way. We use [eclipse-temurin](https://adoptium.net/releases.html?variant=openjdk17&jvmVariant=hotspot) to build the app in Docker, so that's a good choice. [Brew's openjdk@17](https://formulae.brew.sh/formula/openjdk) is also a decent choice.
2. Update your JAVA_HOME to point to your installation.
3. Update your IDE project preferences to use java 17 and its language features. 
![Screen Shot 2022-08-09 at 5 20 30 PM](https://user-images.githubusercontent.com/116679/183763738-54c9ba2a-fd2a-461c-8211-4e9b04abcd93.png)
4. Update your local compose file to use `us.gcr.io/broad-dsp-gcr-public/base/jre:17-debian` as the base image.
5. Test away.


----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment
